### PR TITLE
admin manage user

### LINKED ISSUE #6159: [Manage Users] Manage Users page with search functionality
## Why are we doing this?
The Support Team receives a high volume of inquiries that can only be resolved by software engineers. This adds an extra step in the workflow and increases resolution time for end-users. We need to implement a "Manage Users" page that allows the Support Team to look up SimpleReport users by email address and be able to perform actions on those users, such as undelete, reactivate, or move them to a different organization.

## Action Requested

- Create a "Manage Users" screen that will be accessible only to the Support Team via the Support Admin page. (https://www.simplereport.gov/app/admin)  
- Manage user page should be a copy of the org admin view of a single-user
- The "Manage Users" screen should include a search field at the top of the screen that allows Support Team members to search for SimpleReport users by email. 
- When the search button is clicked, it should hit the endpoint from the related backend ticket and populate the corresponding user info. 
- The search bar should stay at the top of the screen when the screen updates to display the resulting user detail screen. 
- If the existing organization access functionality does not work for Support Team member users, that functionality will be hidden on the User Detail screen, and additional work will be done to address the issue in a separate task. 
- Add back link that matches what we do on other pages ("<- Support admin") and returns the user to the main support admin screen when clicked.

## Acceptance Criteria

- The "Manage Users" page is created and accessible from the Support Admin screen in SimpleReport
- The page includes a search field at the top of the screen that allows support to search for users by email.
- When the search button is clicked, it hits the endpoint from the related backend ticket and populates the corresponding user info.
- The search bar stays at the top of the screen when the screen updates to display the resulting user detail screen.
- If the existing organization admin functionality does not work for Support Team member users, the functionality is hidden.
- A text link at the top of the screen that reads "<- Support admin" is added and returns the user to the main support admin screen when clicked.
- The page displays all the necessary data to render the actions and organization access.
- ~The getUser method is modified to use findByIdIncludeArchived to ensure soft deleted users are also returned in the view.~ (covered in #6158)
- Org access tab is out of scope
- Undelete button is out of scope
- if user is deleted, all actions are greyed out (will handle undelete vs delete button in Undelete user story)
  - be sure to test that reactivate user button is also disabled
- if user doesn't exist in Api User table, display message in middle of screen (see Figma)
- if user exists in Api User table but does not have Okta account, throw error toast and display message in middle of screen (message should prompt reaching out to engineering team)
- if a user is deleted, display "Account deleted" under their name. update existing component for user status to handle status of deleted

## Additional Context

**User view (Manage user)**
This page will serve as the sole page to check and trigger updates to a user account. The page will be linked directly from the support admin menu page (Manager user link). The page will display a search bar at the top that will allow the support admin to search by email. The search bar will have a search button and will validate that the content entered is a valid email address (regex pattern). Upon searching and finding a match the user’s account will load the user view component that displays all the elements shown in the images below.

**Notes**

Using the `setUserIsDeleted` does not remove a user from any groups – undeleting them restores them back to their previous groups

Query to populate this page (Elisa will check) and make sure we have all the necessary data to render the actions and organization access. Elisa Lee
User settings page currently uses `getUser` method to load user information
Data that is returned is:
```json
{
    "data": {
        "user": {
            "id": "4b1af6cf-b696-40ba-a826-ec9988e762ba",
            "firstName": "Test deleted facility",
            "middleName": null,
            "lastName": "User last",
            "roleDescription": "Test-entry user",
            "role": "ENTRY_ONLY",
            "permissions": [
                "UPDATE_TEST",
                "SEARCH_PATIENTS",
                "START_TEST",
                "SUBMIT_TEST"
            ],
            "email": "elisa+testdeletedfacility_dev5@skylight.digital",
            "status": "ACTIVE",
            "organization": {
                "testingFacility": [],
                "__typename": "Organization"
            },
            "__typename": "User"
        }
    }
}
```
However, `getUser` method uses `findById` method which will not return soft deleted users
We may want to create a new getUser method that uses `findByIdIncludeArchived` method to ensure soft deleted users are also returned in this view

**Suggestions**
If an account is in the deleted state the application should not allow for any other updates to it besides undeleting it. We can follow the behavior of the deactivated state, which disables all controls in the user view page.

## References
- **Manage users (Support admin)** [[Google Doc](https://docs.google.com/document/d/1_XAuO35CC3e0aAAaCe1dk2BfPl_agztMBafT1E9GWiw/edit)]
- Figma: Manage User > Search via email
Manage User > Search via email
https://www.figma.com/proto/JU6Uk37PCNhaVs22GZNMVN/SimpleReport---Support-Admin-(2023)?type=design&node-id=16526-3216&t=LPI2c0SKMsXhGkPO-1&scaling=scale-down&page-id=16402%3A26353&starting-point-node-id=16526%3A3216&show-proto-sidebar=1&mode=design

### PR DESCRIPTION
**Pull request by BobanL**
# FRONTEND PULL REQUEST

## Related Issue

- Resolves #6159 

## Changes Proposed

- ~Add a feature flag for the admin manage user page (?)~ ~Will provide Jayna a demo before merge~ View video below
- Add new admin manage user page with the `User Information` tab and new and empty `Organization access` tab.
- Update error message for `RestrictedAccessUserException` to be used for the frontend.
- Add `jest-axe` dependency and add accessibility checks for the new pages.
- In existing modals, run `onClose()` when on the "Continue" option is clicked. 
- Pull out `FacilityAccessTab` and `UserInformationTab` into their own component.
  - Move all modal states into the appropriate tab
- In `ManageUser` replace `UserDetail` with the navigation so the tabs can be customized. 
- In `DemoOktaRepository` return suspended user rather than deprovisioned so the deleted user check works link in `LiveOktaRepository`

## Testing

- ~[Deployed on Dev4](https://dev4.simplereport.gov/app/admin/manage-users)  - Perform regression testing on organization manage user screen
  - Test Admin manage user 
    - Account pending - kihef51477@nmaller.com
    - Account deleted - rilerad238@paldept.com
    - Admin account - your email :) 
    - Valid accounts
      - bawado8153@paldept.com
      - seneh91045@nmaller.com
      - jekes91077@paldept.com
      - nahiti7340@paldept.com
      - remoci7478@royalka.com~
- No longer deployed on dev4 

## Screenshots / Demos

- Video demo
https://github.com/CDCgov/prime-simplereport/assets/10108172/ff8924c9-95b7-478c-b70e-eaaa30b456f7

- Valid account 
image
- Deleted account 
image
- Pending account
image
- Deactivated account 
image
- No user found results
image
- Invalid okta account (admin account)
image

### CODE REVIEW

File: frontend/src/app/Settings/Users/FacilityAccessTab.tsx
Comment (BobanL) [inline frontend/src/app/Settings/Users/FacilityAccessTab.tsx:1]: This mostly comes from the now deleted `UserDetail` file.
View the [actual diff here](https://www.diffchecker.com/e8M8PU8e/)

File: frontend/src/app/Settings/Users/UserDetailUtils.tsx
Comment (BobanL) [inline frontend/src/app/Settings/Users/UserDetailUtils.tsx:1]: This is moved from `UserDetail`. 
See [diff here](https://www.diffchecker.com/text-compare/)


Also, I am not sure if the name of the file is appropriate. If anyone has suggestions let me know!

**Review by fzhao99 — APPROVED** [review]
A true 🤩 effort, great work.

**Comment by mehansen** [comment]
I think instead of a feature flag, what we had said we'd do is record a short demo vid and work with Jayna to share that and get some acknowledgment from support before merging.
doing the feature flag works great, as long as we do the demo vid after and then follow that up with a PR to clean up the feature flag before we mark the ticket as closed

File: frontend/src/app/supportAdmin/pageTitles.ts
Comment (emyl3) [inline frontend/src/app/supportAdmin/pageTitles.ts:6]: ooo I totally forgot this file existed... thank you for adding it here so I know what to update in my PR 😆

File: frontend/src/app/supportAdmin/ManageUsers/UserSearch.tsx
Comment (emyl3) [inline frontend/src/app/supportAdmin/ManageUsers/UserSearch.tsx:29]: based on laura's comment here https://github.com/CDCgov/prime-simplereport/issues/6117#issuecomment-1644291485 -- looks like we want this page to be `Manage users` plural

**Review by johanna-skylight — CHANGES_REQUESTED** [review]
This is really good work @BobanL! the refactoring is super clean and modularized thanks for putting so much though into it. I have left some comments on some edge cases. Let me know if you need more info on those.

File: frontend/src/app/supportAdmin/pageTitles.ts
Comment (johanna-skylight) [reply frontend/src/app/supportAdmin/pageTitles.ts:6]: +1

File: frontend/src/app/Settings/Users/UserDetailUtils.tsx
Comment (emyl3) [reply frontend/src/app/Settings/Users/UserDetailUtils.tsx:19]: Did we also want to handle the `RECOVERY` status as well? I remember in [this doc](https://docs.google.com/document/d/1_XAuO35CC3e0aAAaCe1dk2BfPl_agztMBafT1E9GWiw/edit) we said it would be helpful to surface this but I forget what conclusion we made. 😅 
Screenshot 2023-08-16 at 10 28 07

Comment (johanna-skylight) [reply frontend/src/app/Settings/Users/UserDetailUtils.tsx:19]: I think the group got a consensus that the delay status was a nice to have but extra work that should be done later (specially to not delay this implementation). However, I don't think we ever created a ticket to track it at all. Tagging @mehansen for instructions on how to move forward!.

**Comment by johanna-skylight** [comment]
> I also did some regression testing as an org admin. I noticed that if an org admin deletes a user the `deleted user` still displays for the org admin and the account loads as `active` meaning all the operations are available for the org admin to perform on the user. I have attached screenshots for reference.
> 
> Account was deleted but still displays as an active account. It displays in the sidenav and loads as active in the user information tab. Screenshot 2023-08-16 at 9 33 36 AM

Retested this edge case with Boban and the application worked as expected. Probably I got the users mixed up while testing!.

**Review by emyl3 — APPROVED** [review]
LGTM! Thank you for all your work on this!!

**Comment by fzhao99** [comment]
everything lgtm! will give approval once we get the merge conflicts that I gave to this branch addressed 😅

**Review by johanna-skylight — APPROVED** [review]
Retested in dev4 with site admin and org admin roles and all looks good! Thanks Boban!

**Review by emyl3 — DISMISSED** [review]
Yay!!! Thank you!! 🎉

**Review by johanna-skylight — APPROVED** [review]

**Review by johanna-skylight — APPROVED** [review]

**Review by emyl3 — APPROVED** [review]
🚀

**Pull request merged**
